src/transforms/eigenanalysis.jl

```diff
@@ -2,10 +2,13 @@
 # Licensed under the MIT License. See LICENSE in the project root.
 # ------------------------------------------------------------------
 
+const NDim = Union{Int,Colon}
+
```

Comment:
Please avoid creating these isolated constructs. Simply pass the union type directly to the EigenAnalysis struct or simply don't constrain the type T. EigenAnalysis{T} <:  Transform should be enough.